lib/bolt_ext/schemas/winrm-run_task.json

```diff
@@ -21,7 +21,7 @@
           "description": "Password for SSH transport authentication"
         },
         "port": {
-          "type": "integer",
+          "type": ["integer", "string"],
```

Comment:
Appveyor runs were failing for https://github.com/puppetlabs/bolt/blob/450c17ce617ddfa9c539b5a45786dc3ab74ee574/spec/bolt_ext/server_spec.rb#L159-L178 I am assuming because `JSON.parse` was not preserving the `integer` type for `port`. I am unable to replicate this locally running against vagrant target. Presumably before the JSON validation work `port` being a string did not cause a problem with the test. 

What is a good way to deal with this? Should we coerce all expected integer types after calling JSON.parse and before schema validation? Should we allow both string or integer for port, connection-timeout, etc?